File: solutions/go/difference-of-squares/1/difference_of_squares.go
// Package diffsquares looks squarely at differences.
package diffsquares

// SumOfSquares sums up squares.
func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

// SquareOfSum squares a sum.
func SquareOfSum(n int) int {
	sum := n * (n + 1) / 2
	return sum * sum
}

// Difference looks at differences of squares and squares of differences.
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: solutions/julia/difference-of-squares/4/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
square_of_sum(n) = div(n*(n+1), 2)^2

"Sum the squares of the first `n` positive integers"
sum_of_squares(n) = div(n*(n+1)*(2n+1), 6)

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
difference(n) = square_of_sum(n) - sum_of_squares(n)
